lib/libzfs/libzfs_import.c

```diff
@@ -248,6 +248,17 @@ is_mpath_whole_disk(const char *path)
 	return (wholedisk);
 }
 
+static int
+udev_device_is_ready(struct udev_device *dev)
+{
+#ifdef HAVE_LIBUDEV_UDEV_DEVICE_GET_IS_INITIALIZED
+	return (udev_device_get_is_initialized(dev));
+#else
+	/* wait for DEVLINKS property to be initialized */
+	return (udev_device_get_property_value(dev, "DEVLINKS") != NULL);
```

Comment:
@behlendorf  The long term solution might be to improve the `zpool_label_disk_wait()` to be based on udev monitoring instead of polling a symlink and could eliminate the need for `update_vdev_config_dev_strs()` to check or wait for udev dev initialization.  Perhaps once ZED has the udev monitoring merged we can revisit?